pkg/activator/handler/handler_test.go

```diff
@@ -268,6 +279,168 @@ func TestActivationHandler(t *testing.T) {
+
+	conf1 := config{
+		namespace:   "real-namespace",
+		revName:     "rev-one",
+		serviceName: "rev-one-service",
+		wantBody:    wantBody,
+		// overall max of 2 requests are in the breaker
+		queueDepth:      1,
+		maxConcurrent:   1,
+		initialCapacity: 1,
+		requests:        10,
+		respCh:          respCh,
+	}
+
+	conf2 := config{
+		namespace:   "real-namespace",
+		revName:     "rev-two",
+		serviceName: "rev-two-service",
+		wantBody:    wantBody,
+		// overall max of 20 requests in the Breaker
+		queueDepth:      10,
+		maxConcurrent:   10,
+		initialCapacity: 10,
+		requests:        20,
+		respCh:          respCh,
+	}
+
+	throttler1 := setup(conf1, t)
+	throttler2 := setup(conf2, t)
+	server1 := run(conf1, throttler1, t)
+	server2 := run(conf2, throttler2, t)
+	defer func() {
+		server1.Close()
+		server2.Close()
+	}()
+	assertResponses(wantedSuccess, wantedFailure, overallRequests, respCh, t)
+}
+
+type config struct {
+	namespace       string
+	revName         string
+	serviceName     string
+	requests        int
+	respCh          chan *httptest.ResponseRecorder
+	wantBody        string
+	queueDepth      int32
+	maxConcurrent   int32
+	initialCapacity int32
+}
+
+func setup(conf config, t *testing.T) *activator.Throttler {
+	servingClient := fakeservingclientset.NewSimpleClientset()
+	servingInformerFactory := servinginformers.NewSharedInformerFactory(servingClient, 0*time.Millisecond)
+	revisionInformer := servingInformerFactory.Serving().V1alpha1().Revisions()
+	throttler := activator.NewThrottler(conf.queueDepth, conf.maxConcurrent, conf.initialCapacity, revisionInformer, TestLogger(t))
+	return throttler
+}
+
+func run(conf config, throttler *activator.Throttler, t *testing.T) *httptest.Server {
+	server := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			io.WriteString(w, conf.wantBody)
+		}),
+	)
+
+	act := newStubActivator(conf.namespace, conf.revName, server)
+
+	rt := util.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+		resp, err := http.DefaultTransport.RoundTrip(r)
+		if err != nil {
+			return resp, err
+		}
+		time.Sleep(50 * time.Millisecond)
+		return resp, nil
+	})
+	handler := ActivationHandler{
+		Activator: act,
+		Transport: rt,
+		Logger:    TestLogger(t),
+		Reporter:  &fakeReporter{},
+		Throttler: throttler,
+	}
+
+	for i := 0; i < conf.requests; i++ {
+		go func(index int) {
+			resp := httptest.NewRecorder()
+			req := httptest.NewRequest("POST", "http://example.com", nil)
+			req.Header.Set(activator.RevisionHeaderNamespace, conf.namespace)
+			req.Header.Set(activator.RevisionHeaderName, conf.revName)
+			handler.ServeHTTP(resp, req)
+			conf.respCh <- resp
+		}(i)
+	}
+
+	return server
+}
+
+func assertResponses(wantedSuccess, wantedFailure, overallRequests int, respCh chan *httptest.ResponseRecorder, t *testing.T) {
+	var succeeded int
+	var failed int
+	wantedOverflowMessage := "throttler overload\n"
```

Comment:
If we are going to validate this message is on test we should probably make it a constant in the handler